ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/BasicCallbackHandler.java

```diff
@@ -16,42 +16,33 @@
 package io.confluent.ksql.api.auth;
 
 import java.io.IOException;
-import java.util.Objects;
 import javax.security.auth.callback.Callback;
-import javax.security.auth.callback.CallbackHandler;
 import javax.security.auth.callback.NameCallback;
 import javax.security.auth.callback.PasswordCallback;
 import javax.security.auth.callback.TextOutputCallback;
 import javax.security.auth.callback.UnsupportedCallbackException;
+import org.eclipse.jetty.jaas.callback.DefaultCallbackHandler;
 import org.eclipse.jetty.jaas.callback.ObjectCallback;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class BasicCallbackHandler implements CallbackHandler {
+public class BasicCallbackHandler extends DefaultCallbackHandler {
```

Comment:
Ok, can you say more about why this PR changes from the `javax.security.auth.callback.CallbackHandler` to the `org.eclipse.jetty.jaas.callback.DefaultCallbackHandler`?

Reply:
`DefaultCallbackHandler` is an implementation of `javax.security.auth.callback.CallbackHandler` that "knows" how to handle username/password combos when used in conjunction with `JAASLoginService`:

https://github.com/eclipse/jetty.project/blob/4166f8502635b6f3350d4644cb05c9f8419b4d5e/jetty-jaas/src/main/java/org/eclipse/jetty/jaas/JAASLoginService.java#L189-L205

```java
            CallbackHandler callbackHandler = null;
            if (_callbackHandlerClass == null)
                callbackHandler = new DefaultCallbackHandler();
            else
            {
                Class<?> clazz = Loader.loadClass(_callbackHandlerClass);
                callbackHandler = (CallbackHandler)clazz.getDeclaredConstructor().newInstance();
            }
            
            if (callbackHandler instanceof DefaultCallbackHandler)
            {
                DefaultCallbackHandler dch = (DefaultCallbackHandler)callbackHandler;
                if (request instanceof Request)
                    dch.setRequest((Request)request);
                dch.setCredential(credentials);
                dch.setUserName(username);
            }
```

Using `LoginContext` directly no longer directly works with the `PropertyFileLoginModule` (see https://github.com/eclipse/jetty.project/pull/5518 - requires you to use the `JAASLoginService`) and the `JAASLoginService` doesn't let you pass in instantiated callbacks, it requires passing in just the callback class. That means that I had to take advantage of the code snippet above to get the code properly wired through.

Honestly documentation here is close to impossible to find, so I had to inspect the jetty source code to figure out how to wire this through properly and I very well could be doing something wrong.